src/stringify.ts

```diff
@@ -18,24 +18,29 @@ export function stringify(msgDefs: RosMsgDefinition[]): string {
     }
 
     for (const def of constants) {
-      output += `${def.type} ${def.name} = ${String(def.value)}\n`;
+      output += `${def.type} ${def.name} = ${def.valueText ?? String(def.value)}\n`;
     }
     if (variables.length > 0) {
       if (output.length > 0) {
         output += "\n";
       }
       for (const def of variables) {
         const upperBound = def.upperBound != undefined ? `<=${def.upperBound}` : "";
-        const arrayUpperBound = def.arrayUpperBound != undefined ? `<=${def.arrayUpperBound}` : "";
-        const array = def.isArray === true ? `[${arrayUpperBound}]` : "";
+        const arrayLength =
+          def.arrayLength != undefined
+            ? String(def.arrayLength)
+            : def.arrayUpperBound != undefined
+            ? `<=${def.arrayUpperBound}`
+            : "";
+        const array = def.isArray === true ? `[${arrayLength}]` : "";
```

Comment:
This example from docs might be useful as a unit test: https://docs.ros.org/en/foxy/Concepts/About-ROS-Interfaces.html

```
int32[] unbounded_integer_array
int32[5] five_integers_array
int32[<=5] up_to_five_integers_array

string string_of_unbounded_size
string<=10 up_to_ten_characters_string

string[<=5] up_to_five_unbounded_strings
string<=10[] unbounded_array_of_string_up_to_ten_characters_each
string<=10[<=5] up_to_five_strings_up_to_ten_characters_each
```